### PR TITLE
Tran 103 hasher

### PR DESCRIPTION
**Pull request by amysiv42**
This pull request introduces support for configurable bcrypt salt rounds in the authentication module, enforces validation for salt round limits, and adds corresponding tests. It also refines package metadata and dependencies to support bcrypt functionality.

**Security and configuration improvements:**

* Added support for configurable bcrypt salt rounds via the `BCRYPT_SALT_ROUNDS` environment variable in CI and runtime configuration (`.github/workflows/ci.yml`, `src/auth/src/config/security.ts`). [[1]](diffhunk://#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR32) [[2]](diffhunk://#diff-94a32acc192dc324aff4e5e77d9ef28e87159eff239b7c7deae32c3f5c075ea6R1-R10)
* Introduced salt round limits and validation logic to prevent misconfiguration (`src/auth/src/constants/security.ts`, `src/auth/src/validators/password.ts`, `src/auth/src/types/password.ts`). [[1]](diffhunk://#diff-5999f45f1d111dfc2e94b774bde48006997beafd8b0a9705fceddb9bfe77f448R1-R4) [[2]](diffhunk://#diff-94c0437ea2dc0de739ad13b0e8ef1789edece47055b69a8580007546f22af592R103-R112) [[3]](diffhunk://#diff-9a6552ebb2bc5ca7a9b6d96fd71cf8cb9ed4fc63ae672570e9592179cdbe63c9R27-R32)

**Password hashing functionality:**

* Implemented new password hashing and comparison utilities using `bcryptjs`, with strict salt round validation (`src/auth/src/utils/password/hasher.ts`).
* Added comprehensive unit tests for the hasher utilities, covering edge cases and error handling (`test/auth/utils/hasher.test.ts`).

**Dependency and metadata updates:**

* Updated `package.json` to include `bcryptjs` and its type definitions, and set the author field (`src/auth/package.json`). [[1]](diffhunk://#diff-a7c482496dea0b28451afd9ea64861a00d22c9efc9be2b8a35953b5331986babL6-R6) [[2]](diffhunk://#diff-a7c482496dea0b28451afd9ea64861a00d22c9efc9be2b8a35953b5331986babR15-R21)

### CODE REVIEW

File: src/auth/src/utils/hasher.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import bcrypt from 'bcryptjs';
+
+export async function hasher(str: string): Promise<string> {
```
[the remark this answers is not in the document]
Comment (sairutra) [reply src/auth/src/utils/hasher.ts]: valid if true

File: src/auth/src/utils/hasher.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  const saltRounds = 10;
+  return await bcrypt.hash(str, saltRounds);
+}
+
+export async function compareHash(str: string, hash: string): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (vovashko) [reply src/auth/src/utils/hasher.ts]: valid, please add

**Review by vovashko — CHANGES_REQUESTED** [review]
some little remarks, overall looks fine. add the edge cases to unit tests. You can ask copilot agent in VSCode to add those for you,

File: src/auth/src/utils/hasher.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import bcrypt from 'bcryptjs';
+
+export async function hasher(str: string): Promise<string> {
+  const saltRounds = 10;
+  return await bcrypt.hash(str, saltRounds);
+}
+
```
[the remark this answers is not in the document]
Comment (amysiv42) [reply src/auth/src/utils/hasher.ts]: Better to keep sensitive info in .env

File: src/auth/src/config/security.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { validateSaltLengthLimits } from '../validators/hash.js';
+import { SaltLimits } from '../constants/security.js';
+import { parseIntSave } from '../utils/parse-int-save.js';
+
+
+export function getEnvSaltRounds(saltLimits: typeof SaltLimits) : number {
+  const rounds = parseIntSave(process.env.BCRYPT_SALT_ROUNDS, SaltLimits.DEFAULT_SALT_LENGTH);
```
Comment (vovashko) [inline src/auth/src/config/security.ts]: what is exactly happening here? You import SaltLimits where you defined it as constant object with three parameters. Your function takes SaltLimits as an argument, type is defined in types/password.ts, but you are not importing it. and then you are calling parsing the limits from evironment, and also uses the SaltLimits length from the const?

Comment (amysiv42) [reply src/auth/src/config/security.ts]: No, no it had to be changed. Thanks for pointing out. it is already changed.
